cachito/workers/pkg_managers/gomod.py

```diff
@@ -253,6 +259,70 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
         return {"module": module, "module_deps": module_level_deps, "packages": packages}
 
 
+def _merge_bundle_dirs(root_src_dir, root_dst_dir):
+    """
+    Merge two bundle directories together.
+
+    The contents of root_src_dir will be copied into root_dst_dir, overwriting any files
+    that might already be present. For a description of the algorithm, see
+    https://lukelogbook.tech/2018/01/25/merging-two-folders-in-python/
+
+    In addition to that merge algorithm, however, we also need to make sure that we merge
+    the list file to ensure all versions are represented. In order to protect against merging
+    extra files, we are also checking for the presence of the list.lock file since it should
+    be present according to https://github.com/golang/go/issues/29434
+
+    :param str root_src_dir: the root path to the source directory
+    :param str root_dst_dir: the root path to the destination directory
+    :return: None
+    """
+    for src_dir, dirs, files in os.walk(root_src_dir):
+        dst_dir = src_dir.replace(root_src_dir, root_dst_dir, 1)
+        if not os.path.exists(dst_dir):
+            os.makedirs(dst_dir)
+        for file_ in files:
+            src_file = os.path.join(src_dir, file_)
+            dst_file = os.path.join(dst_dir, file_)
+            if os.path.exists(dst_file):
+                # check to see if we are trying to merge the `list` file
+                # since we have to treat that seperately. We don't want to
+                # delete it or overwrite it -- we need to merge it.
+                if (
+                    file_ == "list"
+                    and os.path.isfile(src_file)
+                    and os.path.exists("{}.lock".format(src_file))
```

Comment:
I wanted to *only* merge the `list` file that we cared about since I didn't know if there were possibly any name collisions that could happen. It sounded to me that the lock file would always be present because they didn't want to introduce synchronization to get it removed. It is a featureful bug... 

I can change this to only check on the file name if desired.

Reply:
Hmm, that's fair. Okay, let's keep it this way :+1: